src/sentry/api/serializers/models/group.py

```diff
@@ -537,13 +538,22 @@ def get_stats(self, item_list, user):
             group_ids = [g.id for g in item_list]
 
             if self.stats_period == "auto":
+                total_period = (self.stats_period_end - self.stats_period_start).total_seconds()
+                rollup = total_period / self.CUSTOM_SEGMENTS
+
+                if rollup > self.CUSTOM_ROLLUP_6H:
+                    rollup = round(rollup / self.CUSTOM_ROLLUP_6H) * self.CUSTOM_ROLLUP_6H
+                elif (2 * rollup) > self.CUSTOM_ROLLUP_6H:
+                    rollup = self.CUSTOM_ROLLUP_6H / 2  # 3hrs
+                elif (3 * rollup) > self.CUSTOM_ROLLUP_6H:
+                    rollup = self.CUSTOM_ROLLUP_6H / 3  # 2hr
+                elif total_period > timedelta(hours=24).total_seconds():
```

Comment:
Should this be an else? 

It seems like we're rounding the rollup to 6, 3 or 2 hours...but in the case of rollups smaller than 2 hours, we only set it to 1 hour if the total period is > 24 hours?

Is there a case where we don't want to just set it to 1 hour? Say a 30 minute rollup over a period shorter than 24 hours?

Reply:
No, if period is < 1d we will divide into 30 buckets, which we do first then I round it to close buckets here for > 1d periods. This is what @adhiraj asked we do